File: py_js_reference/pass_by_value_or_reference.py


def swap(a, b):

    a += 1
    b += 1
    # t = a

    # a = b
    # b = t

    print(f"a = {a}, b = {b}")
    return a, b


def change_arr(arr):
    arrr = arr
    del arrr[:1]


def main():
    arr = [1, 2, 3]
    a = 1
    b = 2

    a, b = swap(a, b)
    change_arr(arr)

    # print(f"a = {a}, b = {b}")
    print(arr)


main()

File: py_js_reference/passByValueOrReference.js
function change_arr(arr) {
  let arrr = JSON.parse(JSON.stringify(arr));
  arrr.splice(0, 1);

  console.log(arrr);
}

function main() {
  let arr = [1, 2, 3];

  change_arr(arr);
  console.log(arr);
}
main();
